examples/demo/piano.html

```diff
@@ -0,0 +1,219 @@
+  <link rel="stylesheet" href="../../build/mathbox.css">
+  <meta name="viewport" content="initial-scale=1, maximum-scale=1">
+</head>
+<body>
+  <script>
+    mathbox = mathBox({
+      plugins: ['core', 'controls', 'cursor'],
+      controls: {
+        klass: THREE.OrbitControls
+      },
+    });
+    three = mathbox.three;
+    three.renderer.setClearColor(new THREE.Color(0x6060A0), 1.0);
+    
+/*
+{DocComment} Piano
+Generates 5 black and 7 white keys of a piano keyboard, and then uses repeat and spread to make multiple
+copies. 
+*/
+   
+   
+    view = mathbox.camera({
+      lookAt: [0,0,0], // changing this doesn't seem to make a difference.
+      position : [0.1,0.9,0.01 ],
+      proxy : true,
+    });
+
+    view.cartesian({
+      range: [[-1, 1], [0, 1], [-1, 1]],
+      scale: [2, 1, 2],
+    });
+
+ 
+    view.grid({
+      width: 5,
+      opacity: 0.5,
+      axes: [1, 3],
+      color: 0x2020A0,
+      blending: 'add'
+    });
+
+    // Emits the four points of one face of a cuboid.
+    // The faces are numbered 0..5.
+    // The cuboid has dimensions w,h,d and is positioned at x,y,z.
+    function cuboid_face( emit, w,h,d, x,y,z, face ){
+      // faces 3,4,5 are displaced relative to their partner 
+      // faces 1,2,3.
+      if( face == 3) x+=w;
+      if( face == 4) y+=h;
+      if( face == 5) z+=d;
+      
+      // It's now as if we are drawing face 0,1 or 2.
+      face = face % 3;
+      // Each face lacks one of the 3 dimensions.
+      // The conditionals ensure we generate 4 points, not 6.
+      // We are just stepping in two of the dimensions.
+      if( face != 0 ) { x+=w;emit(x,y,z); }
+      if( face != 1 ) { y+=h;emit(x,y,z); }
+      if( face != 2 ) { z+=d;emit(x,y,z); }
+      if( face != 0 ) { x-=w;emit(x,y,z); }
+      if( face != 1 ) { y-=h;emit(x,y,z); }
+      if( face != 2 ) { z-=d;emit(x,y,z); }
+    }
+    
+    // The white keys of a piano repeat after 7 
+    // We could set nKeys to 56 here.  Instead we set it low
+    // and then repeat on the GPU.
+    // This example was originally built without nRepeats, and
+    // then adapted to repeat on the GPU.
+    nKeys = 7;     // White keys generated in javascript
+    nRepeats = 8;  // Number of repeats on GPU.
+    // Fewer black keys.
+    nBlackKeys = Math.floor( nKeys * 5.0/7.0 + 0.5);
+    var keyDepth1 = 1.5;  // Distance front edge to black key.
+    var keyDepth2 = 3;    // Distance from start of black key to back.
+    var keyGap = 0.1;     // Gap between keys.
+    
+    // The white keys.
+    view.array({
+      // 12 faces per white key.
+      // (Because we draw each white key as two cuboids).
+      // We are drawing one more face than we need to for each white key.
+      // It is worth it for the convenience of simpler building blocks.
+      length: 12 * nKeys,
+      expr: function (emit, x, t) {
+        r = 2/(nKeys*nRepeats * (1 + keyGap )) ;  // r is the unit of size.
+        p = Math.floor( x );  
```

Comment:
This is unnecessary. Because we're using `array` not `interval` `x` is actually index `i`. The definition of `r` suggests that `interval` might be a better fit.